runtime/near-vm/engine/src/universal/code_memory.rs

```diff
@@ -242,25 +243,25 @@ unsafe impl Send for CodeMemory {}
 /// However it is possible for the mappings inside to grow to accomodate larger code.
 #[derive(Clone)]
 pub struct LimitedMemoryPool {
-    pool: Arc<crossbeam_queue::ArrayQueue<CodeMemory>>,
+    pool: Arc<std::sync::Mutex<Vec<CodeMemory>>>,
 }
 
 impl LimitedMemoryPool {
     /// Create a new pool with `count` mappings initialized to `default_memory_size` each.
     pub fn new(count: usize, default_memory_size: usize) -> rustix::io::Result<Self> {
-        let pool = Arc::new(crossbeam_queue::ArrayQueue::new(count));
-        let this = Self { pool };
+        let pool = Arc::new(std::sync::Mutex::new(Vec::with_capacity(count)));
+        let mut guard = pool.lock().unwrap_or_else(|e| e.into_inner());
         for _ in 0..count {
-            this.pool
-                .push(CodeMemory::create(default_memory_size)?)
-                .unwrap_or_else(|_| panic!("ArrayQueue could not accomodate {count} memories!"));
+            guard.push(CodeMemory::create(default_memory_size)?);
         }
-        Ok(this)
+        drop(guard);
```

Comment:
Tiny nit: to avoid the guard juggling, just create the Vec outside of the mutex and then move the Vec to the mutex